test/tasks/withdraw.test.ts

```diff
@@ -0,0 +1,225 @@
+  tokens: Contract[],
+  trader: Wallet,
+  domainSeparator: TypedDataDomain,
+  settlement: Contract,
+  allowanceManager: Contract,
+  solver: SignerWithAddress,
+) {
+  const encoder = new SettlementEncoder(domainSeparator);
+
+  const consecutiveTokenPairs = tokens.map((token, index) => [
+    token,
+    tokens[(index + 1) % tokens.length],
+  ]);
+  for (const [sell, buy] of consecutiveTokenPairs) {
+    await sell.mint(trader.address, 1);
+    await sell.connect(trader).approve(allowanceManager.address, 1);
+    await encoder.signEncodeTrade(
+      {
+        kind: OrderKind.SELL,
+        partiallyFillable: false,
+        sellToken: sell.address,
+        buyToken: buy.address,
+        sellAmount: 1,
+        buyAmount: 1,
+        feeAmount: 0,
+        validTo: 0xffffffff,
+        appData: 0,
+      },
+      trader,
+      SigningScheme.EIP712,
+    );
+  }
+  const prices: Record<string, number> = {};
+  tokens.forEach((token) => {
+    prices[token.address] = 1;
+  });
+  await settlement.connect(solver).settle(...encoder.encodedSettlement(prices));
+}
+
+describe("Task: withdraw", () => {
+  let deployer: Wallet;
+  let solver: SignerWithAddress;
+  let trader: Wallet;
+  let receiver: Wallet;
+
+  let settlement: Contract;
+  let authenticator: Contract;
+
+  let weth: Contract;
+  let usdc: Contract;
+  let dai: Contract;
+
+  let apiMock: SinonMock;
+  let api: Api;
+
+  beforeEach(async () => {
+    const deployment = await deployTestContracts();
+
+    let solverWallet: Wallet;
+    ({
+      deployer,
+      settlement,
+      authenticator,
+      wallets: [solverWallet, receiver, trader],
+    } = deployment);
+    const foundSolver = (await ethers.getSigners()).find(
+      (signer) => signer.address == solverWallet.address,
+    );
+    expect(foundSolver).not.to.be.undefined;
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    solver = foundSolver!;
+
+    const TestERC20 = await hre.artifacts.readArtifact(
+      "src/contracts/test/TestERC20.sol:TestERC20",
+    );
+    dai = await waffle.deployContract(deployer, TestERC20, ["DAI", 18]);
+    usdc = await waffle.deployContract(deployer, TestERC20, ["USDC", 6]);
+    weth = await waffle.deployContract(deployer, TestERC20, ["WETH", 18]);
+
+    // environment parameter is unused in mock
+    const environment = ("unset environment" as unknown) as Environment;
+    api = new Api("mock", environment);
+    apiMock = sinon.mock(api);
+
+    const { manager, allowanceManager } = deployment;
+    await authenticator.connect(manager).addSolver(solver.address);
+
+    const { chainId } = await ethers.provider.getNetwork();
+    const domainSeparator = domain(chainId, settlement.address);
+    // Trade in order to test automatic retrieval of traded addresses.
+    await tradeTokensForNoFees(
+      [usdc, weth, dai],
+      trader,
+      domainSeparator,
+      settlement,
+      allowanceManager,
+      solver,
+    );
+
+    useDebugConsole();
```

Comment:
why is this needed?

Reply:
I think this is so `console.log` no longer logs for tests.